### PR TITLE
feat(connector): [Klarna] Use connector_request_reference_id as reference to connector

### PR DESCRIPTION
**Pull request by HarshitVashisht11**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [x] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

Included a `merchant_reference1` field in `KlarnaPaymentsRequest` and map it to `connector_request_reference_id` 

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

**Comment by SanchithHegde** [comment]
Hey @HarshitVashisht11, could you comment on #2303 that you've already opened a PR for it so that we can assign it to you? This would help prevent duplicate effort for the same issue.

**Comment by srujanchikke** [comment]
Hey @HarshitVashisht11  ,

Please use the following cargo command -
                                                             
`cargo clippy --all-features` 

ref -[ Cargo commands](https://github.com/juspay/hyperswitch/blob/main/docs/CONTRIBUTING.md#cargo-commands), and fix the errors thrown by clippy for the checks to pass .

Please run `cargo +nightly fmt`  to fix failing formatting check

**Comment by HarshitVashisht11** [comment]
Ok

**Comment by HarshitVashisht11** [comment]
i want help i am not able to find any `payment_id` or `attempt_id` in the file tranformer.rs and klarna.rs

**Comment by srujanchikke** [comment]
> i want help @srujanchikke @SanchithHegde i am not able to find any `payment_id` or `attempt_id` in the file tranformer.rs and klarna.rs

Hey @HarshitVashisht11 ,

This issue is regarding merchant reference id which we pass in connector request , you don't need to handle this in connector response or connector.rs(klarna.rs). If you refer [klarna docs](https://docs.klarna.com/api/payments/#operation/createOrder) , you can find a field called `merchant_reference1 ` . This field needs to be declared in `KlarnaPaymentsRequest`(you can find this in transformers.rs) and map it to `connector_request_reference_id` . If you have any other queries you can ask us in slack ,discord or this thread itself .

**Comment by HarshitVashisht11** [comment]
Hi @SanchithHegde @srujanchikke i have update my code please review my PR

File: crates/router/src/connector/klarna/transformers.rs [hunk 1]
```diff
@@ -157,12 +160,18 @@ impl TryFrom<types::PaymentsResponseRouterData<KlarnaPaymentsResponse>>
...
                 redirection_data: None,
                 mandate_reference: None,
                 connector_metadata: None,
                 network_txn_id: None,
-                connector_response_reference_id: None,
+                connector_response_reference_id: Some(
+                    item.response
+                        .merchant_reference1
+                        .unwrap_or(item.response.order_id),
+                ),
```
Comment (srujanchikke) [inline crates/router/src/connector/klarna/transformers.rs:174]:
```suggestion
                connector_response_reference_id: item.response.merchant_reference1.or(Some(item.response.order_id)),
```
This way we can reduce one operation when merchant_reference1 is not none .

File: crates/router/src/connector/klarna/transformers.rs [hunk 2]
```diff
@@ -140,6 +142,7 @@ impl TryFrom<&KlarnaRouterData<&types::PaymentsAuthorizeRouterData>> for KlarnaP
                         total_amount: i64::from(data.quantity) * (data.amount),
                     })
                     .collect(),
+                merchant_reference1: None,
```
Comment (srujanchikke) [inline crates/router/src/connector/klarna/transformers.rs:145]:
```suggestion
                merchant_reference1: item.router_data.connector_request_reference_id.clone(),
```

File: crates/router/src/connector/klarna/transformers.rs [hunk 3]
```diff
@@ -157,12 +160,18 @@ impl TryFrom<types::PaymentsResponseRouterData<KlarnaPaymentsResponse>>
     ) -> Result<Self, Self::Error> {
         Ok(Self {
             response: Ok(types::PaymentsResponseData::TransactionResponse {
-                resource_id: types::ResponseId::ConnectorTransactionId(item.response.order_id),
+                resource_id: types::ResponseId::ConnectorTransactionId(
```
Comment (srujanchikke) [inline crates/router/src/connector/klarna/transformers.rs:163]: This issue is regarding request , please revert the payment response changes

File: crates/router/src/connector/klarna/transformers.rs [hunk 4]
```diff
@@ -45,12 +45,14 @@ pub struct KlarnaPaymentsRequest {
...
     purchase_country: String,
     purchase_currency: enums::Currency,
+    merchant_reference1: Option<String>,
 }
 
 #[derive(Default, Debug, Deserialize)]
 pub struct KlarnaPaymentsResponse {
     order_id: String,
     fraud_status: KlarnaFraudStatus,
+    merchant_reference1: Option<String>,
```
Comment (srujanchikke) [inline crates/router/src/connector/klarna/transformers.rs:55]: This issue is regarding connector payment request , please revert the response changes .

Comment (HarshitVashisht11) [reply crates/router/src/connector/klarna/transformers.rs:55]: Ok

**Review by srujanchikke — APPROVED** [review]

**Review by SanchithHegde — APPROVED** [review]
Looks good to me!

Thanks for the PR, @HarshitVashisht11!

**Comment by VedantKhairnar** [comment]
Hey @HarshitVashisht11  👋,
Thanks a bunch for all your contributions! We've got a little something for you to show our appreciation.
Just take a moment to fill out this [form](https://forms.gle/yqbtjFuP6oxLaBKN8), and get ready for some awesome swag coming your way.
Thanks!

**Pull request merged**